.ci/pitest-suppressions/pitest-javadoc-suppressions.xml

```diff
@@ -0,0 +1,1973 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mutations>
+  <mutation unstable="false">
+    <sourceFile>JavadocMethodCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck</mutatedClass>
+    <mutatedMethod>beginTree</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.MemberVariableMutator</mutator>
+    <description>Removed assignment to member variable currentClassName</description>
+    <lineContent>currentClassName = &quot;&quot;;</lineContent>
+  </mutation>
+
+  <mutation unstable="false">
+    <sourceFile>AbstractJavadocCheck.java</sourceFile>
+    <mutatedClass>com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck</mutatedClass>
+    <mutatedMethod>validateDefaultJavadocTokens</mutatedMethod>
+    <mutator>org.pitest.mutationtest.engine.gregor.mutators.InlineConstantMutator</mutator>
+    <description>Substituted 2 with 3</description>
+    <lineContent>final String message = String.format(Locale.ROOT,</lineContent>
```

Comment:
I do not understand what is 2 and what is 3 in such code.

```
    private void validateDefaultJavadocTokens() {
        if (getRequiredJavadocTokens().length != 0) {
            final int[] defaultJavadocTokens = getDefaultJavadocTokens();
            Arrays.sort(defaultJavadocTokens);
            for (final int javadocToken : getRequiredJavadocTokens()) {
                if (Arrays.binarySearch(defaultJavadocTokens, javadocToken) < 0) {
                    final String message = String.format(Locale.ROOT,
                            "Javadoc Token \"%s\" from required javadoc "
                                + "tokens was not found in default "
                                + "javadoc tokens list in check %s",
                            javadocToken, getClass().getName());
                    throw new IllegalStateException(message);
                }
            }
        }
    }
```

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html#ROOT

Reply:
I do not either, but taking a conservative guess on what this mutation probably means, here is the decompiled .class file (FernFlower decompiler) related part-
```java
private void validateDefaultJavadocTokens() {                                                                                                                                                                              
    if (this.getRequiredJavadocTokens().length != 0) {                                                                                                                                                                     
        int[] defaultJavadocTokens = this.getDefaultJavadocTokens();                                                                                                                                                       
        Arrays.sort(defaultJavadocTokens);                                                                                                                                                                                 
        int[] var2 = this.getRequiredJavadocTokens();                                                                                                                                                                      
        int var3 = var2.length;                                                                                                                                                                                            
                                                                                                                                                                                                                           
        for(int var4 = 0; var4 < var3; ++var4) {                                                                                                                                                                           
            int javadocToken = var2[var4];                                                                                                                                                                                 
            if (Arrays.binarySearch(defaultJavadocTokens, javadocToken) < 0) {                                                                                                                                             
                String message = String.format(Locale.ROOT, "Javadoc Token \"%s\" from required javadoc tokens was not found in default javadoc tokens list in check %s", javadocToken, this.getClass().getName());        
                throw new IllegalStateException(message);                                                                                                                                                                  
            }                                                                                                                                                                                                              
        }                                                                                                                                                                                                                  
    }                                                                                                                                                                                                                      
                                                                                                                                                                                                                           
}                                                                                                                                                                                                                          
```
Since we used enhanced for loop, I think pit couldn't report this correctly. It inlined some loop variable from `2` to `3` and that test didn't fail.

Reply:
@Vyom-Yadav it is the same idea, it is mutating the value of `javadocToken`

Reply:
> @Vyom-Yadav it is the same idea, it is mutating the value of `javadocToken`

Shouldn't it be reported on [this](https://vyom-yadav.github.io/pit-reports-all/pitest-javadoc/202207100201/com.puppycrawl.tools.checkstyle.checks.javadoc/AbstractJavadocCheck.java.html#org.pitest.mutationtest.report.html.SourceFile@10982139_219) line then?

Reply:
@Vyom-Yadav , please find Junit method that validate message of exception and try to validate whole message.

Reply:
I do not understand your answer, please share here links to Junit methods that validate such exception

Reply:
@romani Yes, it will fail if we remove the first word from the message, but how is it implying the mutation we are talking about right now?
`Substituted 2 with 3`

Reply:
Please contact pitest maintainer and ask them a question what it could mean.

Reply:
Please remember, pitest is a bytecode manipulator. It doesn't read or understand the source code. If you really want to understand what pitest is doing, you need to look at the bytecode.

> I do not understand what is 2 and what is 3 in such code.

The source,
````
                    final String message = String.format(Locale.ROOT,
                            "Javadoc Token \"%s\" from required javadoc "
                                + "tokens was not found in default "
                                + "javadoc tokens list in check %s",
                            javadocToken, getClass().getName());
````

converts to:

> javap -c -l '.\AbstractJavadocCheck.class'

````
      49: getstatic     #32                 // Field java/util/Locale.ROOT:Ljava/util/Locale;
      52: ldc           #42                 // String Javadoc Token \"%s\" from required javadoc tokens was not found in default javadoc tokens list in check %s
      54: iconst_2
      55: anewarray     #34                 // class java/lang/Object
      58: dup
      59: iconst_0
      60: iload         5
      62: invokestatic  #14                 // Method java/lang/Integer.valueOf:(I)Ljava/lang/Integer;
      65: aastore
      66: dup
      67: iconst_1
      68: aload_0
      69: invokevirtual #36                 // Method java/lang/Object.getClass:()Ljava/lang/Class;
      72: invokevirtual #37                 // Method java/lang/Class.getName:()Ljava/lang/String;
      75: aastore
      76: invokestatic  #38                 // Method java/lang/String.format:(Ljava/util/Locale;Ljava/lang/String;[Ljava/lang/Object;)Ljava/lang/String;
      79: astore        6
````

I'll avoid going too far, but basically pitest is manipulating the iconst_2 ( https://github.com/hcoles/pitest/blob/master/pitest/src/main/java/org/pitest/mutationtest/engine/gregor/mutators/InlineConstantMutator.java#L204 ) connected to the following lines:
````
      54: iconst_2
      55: anewarray     #34                 // class java/lang/Object
````

So what is this code doing from and how did it come from the source?

Basically, `String.format` parameters are `format(Locale l, String format, Object... args)` and this all comes down to how `Object...` is handled in the bytecode when calling this method.

The source passes each part of the `args` as separate strings, but the method doesn't accept separate strings. It accepts only an array. What Java does for the developers to make coding easier is it sees and makes this simple conversion for you.

To make this conversion it actually instantiates an Object array for you and puts the arguments you are passing into the array and passes that to the method. In this case we are passing **2** strings to the `args`. So basically your simple `   javadocToken, getClass().getName()` in the source actually becomes `new Object[] {javadocToken, getClass().getName()}` or basically:
````
final Object[] temp = new Object[2];
temp[0] = javadocToken;
temp[1] = getClass().getName();
````

You all can start to see where the 2 comes into play here. It is the number of arguments being passed specifically to the `Object...`.

If you actually embed this change in the source and place it on an isolated line and run pitest, you will see the surviving mutant is only on this line and the rest of the code passes.

Reply:
We can not use this mutation, it this specific case can not be excluded by config

Reply:
Please create issue and move comment of rnveach to it , keep suppression and put to new issue obove it to explain reason.

Reply:
Done, created at #11895